src/compiler/factory.ts

```diff
@@ -1618,17 +1618,23 @@ namespace ts {
         );
     }
 
-    function createReactNamespace(reactNamespace: string, parent: JsxOpeningLikeElement) {
+    export function createJsxFactory(jsxFactory: string) {
+        // No explicit validation of this parameter is required. Users are

```

Comment:
I went with this approach for simplicity, based on the belief this was acceptable behaviour as per @RyanCavanaugh's [comment](https://github.com/Microsoft/TypeScript/issues/9582#issuecomment-250320276):

> No explicit validation of this parameter occurs; users are assumed to have provided a correct string